services/device-registry/pom.xml

```diff
@@ -0,0 +1,190 @@
+      <groupId>org.eclipse.hono</groupId>
+      <artifactId>hono-demo-certs</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-logging</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-autoconfigure</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <!--
+          Copy legal documents from "legal" module to "target/classes" folder
+          so that we make sure to include legal docs in all modules.
+         -->
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <!--
+          Execution and configuration for copying certificates from related module
+          to "target/classes" folder so that we can include them in the image.
+         -->
+        <executions>
+          <execution>
+            <id>copy_demo_certs</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>unpack-dependencies</goal>
+            </goals>
+            <configuration>
+              <includeArtifactIds>hono-demo-certs</includeArtifactIds>
+              <outputDirectory>${project.build.outputDirectory}/certs</outputDirectory>
+              <excludes>META-INF/**</excludes>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>build-docker-image</id>
+      <activation>
+        <property>
+          <name>docker.host</name>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-maven-plugin</artifactId>
+          </plugin>
+          <plugin>
+            <groupId>io.fabric8</groupId>
+            <artifactId>docker-maven-plugin</artifactId>
+            <configuration>
+              <verbose>true</verbose>
+              <images>
+                <image>
+                  <build>
+                    <from>${java-base-image.name}</from>
+                    <ports>
+                      <port>5671</port>
+                      <port>5672</port>
+                    </ports>
+                    <runCmds>
+                      <runCmd>adduser -S -D hono nogroup</runCmd>
+                    </runCmds>
+                    <user>hono</user>
+                    <workdir>/home/hono</workdir>
+                    <cmd>
+                      <exec>
+                        <arg>java</arg>
+                        <arg>-Dvertx.disableDnsResolver=true</arg>
```

Comment:
you can safely remove these three lines regarding DNS. We are now using the vert.x hostname resolver (which needs to be configured explicitly, though). See `ApplicationConfing` from the hono-app.